File: src/test/java/com/d20charactersheet/framework/boc/service/exportimport/ImportCharacterV1XmlTest.java
package com.d20charactersheet.framework.boc.service.exportimport;

import static org.junit.Assert.assertEquals;

import javax.xml.parsers.DocumentBuilder;
import javax.xml.parsers.DocumentBuilderFactory;

import org.junit.Before;
import org.junit.Test;
import org.w3c.dom.Document;
import org.w3c.dom.Element;

import com.d20charactersheet.framework.DnDv35Universe;
import com.d20charactersheet.framework.Universe;
import com.d20charactersheet.framework.boc.model.Race;
import com.d20charactersheet.framework.boc.service.GameSystem;

public class ImportCharacterV1XmlTest {

  private ImportCharacterV1Xml importCharacter;
  private Document document;

  @Before
  public void setUp() throws Exception {
    final Universe universe = new DnDv35Universe();
    final DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
    final DocumentBuilder builder = factory.newDocumentBuilder();
    document = builder.newDocument();
    final GameSystem gameSystem = universe.getGameSystem();
    final ImportContext importContext = new ImportContext();
    importContext.setGameSystemName(gameSystem.getName());
    importContext.setAllRaces(gameSystem.getAllRaces());
    importContext.setAllClasses(gameSystem.getAllCharacterClasses());
    importContext.setAllXpTables(gameSystem.getAllXpTables());
    importContext.setAllSkills(gameSystem.getAllSkills());
    importContext.setAllFeats(gameSystem.getAllFeats());
    importContext.setAllWeapons(gameSystem.getAllWeapons());
    importContext.setAllArmor(gameSystem.getAllArmor());
    importContext.setAllGoods(gameSystem.getAllGoods());
    importContext.setAllSpells(gameSystem.getAllSpells());
    importContext.setAllSpelllists(gameSystem.getAllSpelllists());
    importContext.setAllAbilities(gameSystem.getAllAbilities());
    importCharacter = new ImportCharacterV1Xml(importContext, document);
  }

  @Test
  public void testGetRace() throws Exception {
    final Element raceElement = document.createElement(XmlConstants.TAG_RACE);
    raceElement.setTextContent("High Elf");

    final Race race = importCharacter.new ImportAppearance().getRace(raceElement);

    assertEquals(2, race.getId());
    assertEquals("High Elf", race.getName());
  }

}

File: src/test/java/com/d20charactersheet/framework/boc/service/exportimport/ImportCharacterV1XmlTest.kt
package com.d20charactersheet.framework.boc.service.exportimport

import com.d20charactersheet.framework.DnDv35Universe
import com.d20charactersheet.framework.Universe
import org.junit.Assert
import org.junit.Before
import org.junit.Test
import org.w3c.dom.Document
import javax.xml.parsers.DocumentBuilderFactory

class ImportCharacterV1XmlTest {

    private lateinit var importCharacter: ImportCharacterV1Xml
    private lateinit var document: Document

    @Before
    @Throws(Exception::class)
    fun setUp() {
        val universe: Universe = DnDv35Universe()
        val factory = DocumentBuilderFactory.newInstance()
        val builder = factory.newDocumentBuilder()
        document = builder.newDocument()
        val gameSystem = universe.gameSystem
        val importContext = ImportContext()
        importContext.gameSystemName = gameSystem.name
        importContext.allRaces = gameSystem.allRaces
        importContext.allClasses = gameSystem.allCharacterClasses
        importContext.allXpTables = gameSystem.allXpTables
        importContext.allSkills = gameSystem.allSkills
        importContext.allFeats = gameSystem.allFeats
        importContext.allWeapons = gameSystem.allWeapons
        importContext.allArmor = gameSystem.allArmor
        importContext.allGoods = gameSystem.allGoods
        importContext.allSpells = gameSystem.allSpells
        importContext.allSpelllists = gameSystem.allSpelllists
        importContext.allAbilities = gameSystem.allAbilities
        importCharacter = ImportCharacterV1Xml(importContext, document)
    }

    @Test
    fun testGetRace() {
        val raceElement = document.createElement(XmlConstants.TAG_RACE)
        raceElement.textContent = "High Elf"
        val race = importCharacter.ImportAppearance().getRace(raceElement)
        Assert.assertEquals(2, race.id.toLong())
        Assert.assertEquals("High Elf", race.name)
    }
}